File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/1018_체스판다시칠하기/1018.py
N, M = map(int, input().split())
map = [input() for _ in range(N)]

def count_diff(row, col, pattern):
    cnt = 0
    for i in range(row, row + 8):
        for j in range(col, col + 8):
            if map[i][j] != pattern[(i + j) % 2]:
                cnt += 1
    return cnt

ans = N * M
for i in range(N - 7):
    for j in range(M - 7):
        cnt_BW = count_diff(i, j, "BW")
        cnt_WB = count_diff(i, j, "WB")
        ans = min(ans, min(cnt_BW, cnt_WB))

print(ans)

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/1018_체스판다시칠하기/1018.cpp
#include <stdio.h>
#include <algorithm>
using namespace std;

int N, M;
char map[51][51];

int count_diff(int row, int col, char *pattern)
{
    int cnt = 0;
    for (int i = row; i < row + 8; i++)
        for (int j = col; j < col + 8; j++)
            if (map[i][j] != pattern[(i + j) % 2])
                cnt++;
    return cnt;
}

int main()
{
    scanf("%d %d", &N, &M);
    for (int i = 0; i < N; i++)
        scanf("%s", map[i]);

    int ans = N * M;
    for (int i = 0; i <= N - 8; i++)
    {
        for (int j = 0; j <= M - 8; j++)
        {
            int cnt_BW = count_diff(i, j, "BW");
            int cnt_WB = count_diff(i, j, "WB");
            ans = min(ans, min(cnt_BW, cnt_WB));
        }
    }
    
    printf("%d\n", ans);
    return 0;
}
